Src/FluentAssertions/Collections/MaximumMatching/MaximumMatchingSolver.cs

```diff
@@ -79,7 +79,7 @@ private IEnumerable<Match> FindMatchForPredicate(Predicate<TValue> predicate, Ma
         return Enumerable.Empty<Match>();
     }
 
-    private IEnumerable<Element<TValue>> GetMatchingElements(Predicate<TValue> predicate)
+    private List<Element<TValue>> GetMatchingElements(Predicate<TValue> predicate)
```

Comment:
I actually dislike the changes in this PR. I often use `IEnumerable<T>` to prevent the call side from modifying the collection.

Reply:
Did you notice that all changed methods are `private`?
If that doesn't make the situation better, then I'll just close the PR again, no worries.

Reply:
That's a good point. Will that CA1859 also complain about `protected`, `internal` and `public` methods like that?

Reply:
I'll have to check up on that.

Reply:
It can be configured via `.editorconfig`
> CA1859: default value is private.

https://github.com/dotnet/roslyn-analyzers/blob/main/docs/Analyzer%20Configuration.md#editorconfig-format